packages/wallet-backend-update/src/account/service.ts

```diff
@@ -0,0 +1,200 @@
+  rapyd: RapydClient
+  rafiki: RafikiClient
+  logger: Logger
+}
+
+export class AccountService implements IAccountService {
+  constructor(private deps: CountriesServiceDependencies) {}
+
+  public async createAccount(
+    userId: string,
+    name: string,
+    assetId: string
+  ): Promise<Account> {
+    const existingAccount = await Account.query()
+      .where('userId', userId)
+      .where('name', name)
+      .first()
+    if (existingAccount) {
+      throw new Conflict(`An account with the name '${name}' already exists`)
+    }
+
+    const asset = await this.deps.rafiki.getAssetById(assetId)
+
+    if (!asset) {
+      throw new NotFound()
+    }
+
+    const existingAssetAccount = await Account.query()
+      .where('assetCode', asset.code)
+      .where('userId', userId)
+      .first()
+
+    if (existingAssetAccount) {
+      throw new Conflict(
+        `An account with the same asset ${asset.code} already exists`
+      )
+    }
+
+    // issue virtual account to wallet
+    const user = await User.query().findById(userId)
+    if (!user) {
+      throw new NotFound()
+    }
+
+    const result = await this.deps.rapyd.issueVirtualAccount({
+      country: user.country ?? '',
+      currency: asset.code,
+      ewallet: user.rapydWalletId ?? ''
+    })
+
+    if (result.status.status !== 'SUCCESS') {
+      throw new Error(
+        `Unable to issue virtal account to ewallet: ${result.status.message}`
+      )
+    }
+
+    // save virtual bank account number to database
+    const virtualAccount = result.data
+
+    const account = await Account.query().insert({
+      name,
+      userId,
+      assetCode: asset.code,
+      assetId,
+      virtualAccountId: virtualAccount.id
+    })
+
+    await this.deps.rapyd.simulateBankTransferToWallet({
+      amount: 0,
+      currency: account.assetCode,
+      issued_bank_account: account.virtualAccountId
+    })
+
+    if (!user || !user.rapydWalletId) {
+      throw new NotFound()
+    }
+
+    const accountsBalance = await this.deps.rapyd.getAccountsBalance(
+      user.rapydWalletId
+    )
+    account.balance = formatBalance(
+      accountsBalance.data.find((acc) => acc.currency === account.assetCode)
+        ?.balance ?? 0
+    )
+
+    return account
+  }
+
+  public async getAccounts(userId: string): Promise<Account[]> {
+    const accounts = await Account.query().where('userId', userId)
+
+    const user = await User.query().findById(userId)
+
+    if (!user || !user.rapydWalletId) {
+      throw new NotFound()
+    }
+
+    const accountsBalance = await this.deps.rapyd.getAccountsBalance(
+      user.rapydWalletId
+    )
+
+    accounts.forEach((acc) => {
+      acc.balance = formatBalance(
+        accountsBalance.data.find(
+          (rapydAccount) => rapydAccount.currency === acc.assetCode
+        )?.balance ?? 0
+      )
+    })
+
+    return accounts
+  }
+
+  public async getAccountById(
+    userId: string,
+    accountId: string
```

Comment:
I wouldn't necessarily go for params sent in objects for all methods, especially when only 2-3 params are sent. Linter does a good job of always infering types, no need for extra defined Types of objects that need to be sent as params.

However, I do agree that it would be nice to have the same style everywhere, so I will wait for @dragosp1011  and @rico191013 's opinion on this so that we decide

Reply:
agree with Beni, for more than 2-3 params we can use object. If all the params are like a logical structure (a payment payload for example) then it can be an object even if there are 2-3 params.